components/google_drive/google_drive.app.js

```diff
@@ -95,9 +95,62 @@ module.exports = {
         newStartPageToken,
       };
     },
-    async getPageToken() {
+    async getPageToken(driveId) {
       const drive = this.drive();
-      return (await drive.changes.getStartPageToken({})).data.startPageToken;
+      const request = driveId ? {
+        driveId,
+        supportsAllDrives: true,
+      } : {};
+      const { data } = await drive.changes.getStartPageToken(request);
+      return data.startPageToken;
+    },
+    checkHeaders(headers, subscription, channelID) {
+      if (
+        !headers["x-goog-resource-state"] ||
+        !headers["x-goog-resource-id"] ||
+        !headers["x-goog-resource-uri"] ||
+        !headers["x-goog-message-number"]
+      ) {
+        console.log("Request missing necessary headers: ", headers);
+        return false;
+      }
+
+      const incomingChannelID = headers["x-goog-channel-id"];
+      if (incomingChannelID !== channelID) {
+        console.log(
+          `Channel ID of ${incomingChannelID} not equal to deployed component channel of ${channelID}`
+        );
+        return false;
+      }
+
+      if (headers["x-goog-resource-id"] !== subscription.resourceId) {
+        console.log(
+          `Resource ID of ${resourceId} not currently being tracked. Exiting`
```

Comment:
`subscription.resourceId`? I don't think `resourceId` is in scope here.

Reply:
Good catch